File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    d = dict().fromkeys(id_list, []) # k = 유저, v = 신고한 u
    report = set(report)
    tmp = [list(x) for x in zip(*[x.split() for x in report])][1]
    cnt = [tmp.count(x) for x in id_list] # 신고 건
    
    for r in report:
        id1, id2 = r.split()
        if cnt[id_list.index(id2)] >= k:
            if d[id1]:
                d[id1].append(id2)
            else:
                d[id1] = [id2]
    return [len(d[x]) for x in id_list]

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length]; // 0으로 초기화
        Map <String, Integer> index = new HashMap();
        Map <String, Set<String>> log = new HashMap();
        for (int i = 0; i < id_list.length; i++){ // ans index 정보
            index.put(id_list[i], i);
            log.put(id_list[i], new HashSet());
        }
        
        String[] tmp;
        Set<String> values;
        for (int i = 0; i < report.length; i++){
            tmp = report[i].split(" ");
            if (!tmp[0].equals(tmp[1])) {
                values = log.get(tmp[1]);
                values.add(tmp[0]);
                log.put(tmp[1], values);
            }
        }
        
        for (int i = 0; i < id_list.length; i++){
            if (log.get(id_list[i]).size() >= k) {
                for (String name: log.get(id_list[i])){
                    answer[index.get(name)]++;
                }
            }
        }
        
        
        return answer;
    }
}